src/Native/Runtime/arm/StubDispatch.S

```diff
@@ -2,81 +2,161 @@
+	        // we may A/V on a null this. Both of these suggest we need a real prolog and epilog.
+          PROLOG_PUSH {r1-r2}
+
+          // load r11 to point to the cache block.
+          ldr     r2, [r12, #OFFSETOF__InterfaceDispatchCell__m_pCache]
+
+          // Load the EEType from the object instance in r0.
+          ldr     r1, [r0]
+
+          CurrentOffset = OFFSETOF__InterfaceDispatchCache__m_rgEntries
+          // For each entry in the cache, see if its EEType type matches the EEType in rax.
+          // If so, call the second cache entry.  If not, skip the InterfaceDispatchCacheEntry.
+          .rept \entries
+              ldr     r3, [r2, #CurrentOffset]
+              cmp     r1, r3
+              bne     0f
+              ldr     r3, [r2, #(CurrentOffset + 4)]
+              blx     r3
+          0:
+              CurrentOffset = CurrentOffset + 8
+          .endr
+
+          // r12 still contains the the indirection cell address.
+
+          bl           C_FUNC(RhpInterfaceDispatchSlow)
+LEAF_END RhpInterfaceDispatch\entries, _TEXT
+
+.endm // DEFINE_INTERFACE_DISPATCH_STUB
+
+// Define all the stub routines we currently need.
+//
+// The mrt100dbi requires these be exported to identify mrt100 code that dispatches back into managed.
+// If you change or add any new dispatch stubs, please also change slr.def and dbi\process.cpp CordbProcess::GetExportStepInfo
+//
+DEFINE_INTERFACE_DISPATCH_STUB 1
+DEFINE_INTERFACE_DISPATCH_STUB 2
+DEFINE_INTERFACE_DISPATCH_STUB 4
+DEFINE_INTERFACE_DISPATCH_STUB 8
+DEFINE_INTERFACE_DISPATCH_STUB 16
+DEFINE_INTERFACE_DISPATCH_STUB 32
+DEFINE_INTERFACE_DISPATCH_STUB 64
 
 // Stub dispatch routine for dispatch to a vtable slot
 LEAF_ENTRY RhpVTableOffsetDispatch, _TEXT
-  // UNIXTODO: Implement this function
-  // int 3
+        // On input we have the indirection cell data structure in r12. But we need more scratch registers and
+        // we may A/V on a null this. Both of these suggest we need a real prolog and epilog.
+        PROLOG_PUSH {r1}
+
+        // r12 currently holds the indirection cell address. We need to update it to point to the vtable
+        // offset instead.
+        ldr     r12, [r12, #OFFSETOF__InterfaceDispatchCell__m_pCache]
+
+        // Load the EEType from the object instance in r0.
+        ldr     r1, [r0]
+
+        // add the vtable offset to the EEType pointer
+        add     r12, r1, r12
+
+        // Load the target address of the vtable into r12
+        ldr     r12, [r12]
+
+        EPILOG_POP {r1}
+        EPILOG_BRANCH_REG r12
 LEAF_END RhpVTableOffsetDispatch, _TEXT
 
 // Initial dispatch on an interface when we don't have a cache yet.
 LEAF_ENTRY RhpInitialInterfaceDispatch, _TEXT
-ALTERNATE_ENTRY RhpInitialDynamicInterfaceDispatch
-
-  // Just tail call to the cache miss helper.
-  // jmp     C_FUNC(RhpInterfaceDispatchSlow)
-
+          // Just tail call to the cache miss helper.
+          bl           C_FUNC(RhpInterfaceDispatchSlow)
```

Comment:
On Windows ARM, this method pops r12: https://github.com/dotnet/corert/blob/master/src/Native/Runtime/arm/StubDispatch.asm#L174 . Not sure whether this difference is intentional or not.

Could you please describe the calling conventions that you are doing to use for the interface stub dispatch calls? Is it going to be the same as Windows ARM CoreRT conventions, or is it going to be different.